src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableHashSet_1.cs

```diff
@@ -754,9 +754,14 @@ private static bool Overlaps(IEnumerable<T> other, MutationInput origin)
         private static bool SetEquals(IEnumerable<T> other, MutationInput origin)
         {
             Requires.NotNull(other, nameof(other));
+            ISet<T>? otherSet = other switch
+            {
+                ImmutableHashSet<T> immutableHashSet => immutableHashSet.WithComparer(origin.EqualityComparer),
+                HashSet<T> hashSet when origin.EqualityComparer == hashSet.Comparer => other as HashSet<T>,
```

Comment:
Return the matched variable `hashSet` instead of using `as` in this branch, then `otherSet` can be non-nullable and the `?.` will be unnecessary.

Reply:
Makes sense, updated.